exercises/react/test/test_react.c

```diff
@@ -0,0 +1,110 @@
+#include <stddef.h>
+#include <stdio.h>
+#include "vendor/unity.h"
+#include "../src/react.h"
+
+void test_input_cells_have_value(void)
+{
+   struct reactor *r = create_reactor();
+   struct cell *input = create_input_cell(r, 2);
+
+   TEST_ASSERT_EQUAL_INT(2, cell_value(input));
+
+   free_reactor(r);
+}
+
+static int plus1(int x)
+{
+   return x + 1;
+}
+
+void test_compute_cells_calculate_initial_value(void)
+{
+   struct reactor *r = create_reactor();
+   struct cell *input = create_input_cell(r, 1);
+   struct cell *output = create_compute1_cell(r, input, plus1);
+
+   TEST_ASSERT_EQUAL_INT(2, cell_value(output));
+
+   free_reactor(r);
+}
+
+void test_compute_cells_update_value_when_dependencies_are_changed(void)
+{
+   struct reactor *r = create_reactor();
+   struct cell *input = create_input_cell(r, 1);
+   struct cell *output = create_compute1_cell(r, input, plus1);
+
+   cell_value_set(input, 3);
+   TEST_ASSERT_EQUAL_INT(4, cell_value(output));
+
+   free_reactor(r);
+}
+
+static int times2(int x)
+{
+   return x * 2;
+}
+
+static int times30(int x)
+{
+   return x * 30;
+}
+
+static int plus(int x, int y)
+{
+   return x + y;
+}
+
+// TODO: any better way to represent these?
```

Comment:
In particular: This is going to hurt every time I need to use this cb1, since I need to reset the value of `cb_1_called` and `cb1_last`. Plus, there is the test where I will need to have a total of three callbacks. https://github.com/exercism/x-common/blob/master/exercises/react/canonical-data.json#L284

Having to have three sets of globals + function to manipulate them seems too much. I should perhaps allow a `void *` to be passed into each `add_callback`, and the API will call the callback function passing the `void *` back in.